sklearn/linear_model/_stochastic_gradient.py

```diff
@@ -1032,9 +1058,17 @@ def predict_log_proba(self):
 
         See ``predict_proba`` for details.
 
+        Side note: ``predict_log_proba`` is a property. Technically, it
+        accepts no arguments except self, and returns a method,
+        ``_predict_log_proba``. However, since it returns a callable
+        method, when it's invoked in code, it appears
+        like it's being called with an argument (of input data) X, and
+        returning probabilities T.
+
         Parameters
         ----------
-        X : array-like, shape (n_samples, n_features)
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
```

Comment:
```suggestion
        X : {array-like, sparse matrix} of shape (n_samples, n_features)
```

Reply:
Made this change, although it's inconsistent with other similar docstrings in the file (like [this](https://github.com/scikit-learn/scikit-learn/blob/968252dfb4c252e252614ae8736cc6b1ebdde9ee/sklearn/linear_model/_stochastic_gradient.py#L1209)).